kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsServiceImpl.java

```diff
@@ -0,0 +1,698 @@
+            }
+        }
+//        getBusinessObjectService().save(updatedVDEList);
+//        getBusinessObjectService().save(noteList);
+        dataFileStream.flush();
+        dataFileStream.close();
+        writeVendorCountFile(outputVendorLineList.size());
+        KualiInteger listSize = new KualiInteger(outputVendorLineList.size());
+        saveLastRunDate(jobRunDate, listSize);
+        return true;
+    }
+
+    /**
+     * write the file containing number of vendors processed for ecustoms
+     *
+     * @param vendorDetails
+     */
+    private void writeVendorCountFile(int size) throws IOException {
+        FileOutputStream countFileStream = new FileOutputStream(vendorCountFile);
+        String outputMessage = "Vendors added to ecustoms file = " + size;
+        LOG.info(outputMessage);
+        countFileStream.write(outputMessage.getBytes());
+        countFileStream.flush();
+        countFileStream.close();
+    }
+
+    /**
+     * Get the list of vendors that need to be processed since the last daily run.
+     *
+     * @param jobRunDate
+     * @return list of vendors to process for security compliance
+     * @throws WorkflowException
+     */
+    private List<VendorDetail> getDailyVendorDetailList(Date jobRunDate) {
+        List<VendorDetail> retval = new ArrayList<VendorDetail>();
+
+        List<VendorDetail> vendorDetails = new ArrayList<VendorDetail>();
+
+        List<VendorDetail> recentVendorList = getRecentVendorList();
+        List<VendorDetail> exportList = getVendorsWithExportFlagSetToNo();
+
+        vendorDetails.addAll(recentVendorList);
+        vendorDetails.addAll(exportList);
+
+        if (vendorDetails.isEmpty()) {
+            LOG.info("No vendor candidates found.");
+            return retval;
+        } else {
+            LOG.info("Found " + vendorDetails.size() + " total vendor candidates for ecustoms.");
+        }
+
+        for (VendorDetail vendorDetail : vendorDetails) {
+            boolean wasProcessedToday = hasProcessedToday(vendorDetail);
+            if (!wasProcessedToday) {
+                retval.add(vendorDetail);
+            }
+        }
+        // retval.addAll(vendorDetails);
+        LOG.info("Found " + retval.size() + " vendors to be processed by batch for ecustoms.");
+        return retval;
+
+    }
+
+    /**
+     * Get a list of new vendors that are ready to be processed.
+     *
+     * @param newVendorNotes
+     * @return List of VendorDetail objects
+     */
+    private List<VendorDetail> getRecentVendorList() {
+        List<VendorDetail> retval = new ArrayList<VendorDetail>();
+
+        List<DocumentRouteHeaderValue> docHeaderList = getRecentFinalizedVendorsDocHeaders();
+
+        for (DocumentRouteHeaderValue routeHeader : docHeaderList) {
+            String documentId = routeHeader.getDocumentId();
+            MaintenanceDocument doc = null;
+            LOG.info("+================================================================================");

```

Comment:
Unless this is going to the specific job's log, I'd reconsider throwing in decoration. If it is a job's log, go nuts to your heart's content.